src/main/java/com/github/dockerjava/api/model/AuthConfig.java

```diff
@@ -30,48 +29,52 @@
     @JsonProperty("serveraddress")
     private String serverAddress = DEFAULT_SERVER_ADDRESS;
 
+    @JsonProperty("auth")
     private String auth;
 
     public String getUsername() {
         return username;
     }
 
-    public void setUsername(String username) {
+    public AuthConfig withUsername(String username) {
         this.username = username;
+        return this;
     }
 
     public String getPassword() {
         return password;
     }
 
-    public void setPassword(String password) {
+    public AuthConfig withPassword(String password) {
         this.password = password;
+        return this;
     }
 
     public String getEmail() {
         return email;
     }
 
-    public void setEmail(String email) {
+    public AuthConfig withEmail(String email) {
         this.email = email;
+        return this;
     }
 
     public String getServerAddress() {
         return serverAddress;
     }
 
-    public void setServerAddress(String serverAddress) {
+    public AuthConfig withServerAddress(String serverAddress) {
```

Comment:
Not sure if this should be included in this PR but I suggest to rename method to `withRegistryUrl`. WDYT?